src/components/Table/GridTable.test.tsx

```diff
@@ -379,6 +379,134 @@ describe("GridTable", () => {
       expect(sortHeader_1()).toBeUndefined();
     });
 
+    it("can have primary rows when sorting", async () => {
+      // Given the table is using client-side sorting
+      type FavoriteData = { name: string; value: number; favorite: boolean };
+      type FavoriteRow = SimpleHeaderAndData<FavoriteData>;
+
+      const favNameColumn: GridColumn<FavoriteRow> = { header: () => "Name", data: ({ name }) => name };
+      const favValueColumn: GridColumn<FavoriteRow> = { header: () => "Value", data: ({ value }) => value };
+      const favoriteColumn: GridColumn<FavoriteRow> = { header: () => "Favorite", data: ({ favorite }) => favorite };
+
+      const r = await render(
+        <GridTable
+          columns={[favNameColumn, favValueColumn, favoriteColumn]}
+          sorting={{ on: "client", primary: [favoriteColumn, "DESC"] }}
+          rows={[
+            simpleHeader,
+            // And the data is initially unsorted
+            { kind: "data", id: "2", data: { name: "b", value: 2, favorite: true } },
+            { kind: "data", id: "1", data: { name: "a", value: 3, favorite: false } },
+            { kind: "data", id: "3", data: { name: "c", value: 1, favorite: false } },
+          ]}
+        />,
+      );
+      // Then the data is initially render sorted by 1st column
+      expect(cell(r, 1, 0)).toHaveTextContent("b");
+
+      // And when sorted by column 1
+      click(r.sortHeader_0);
+      // Then 'name: b' row is first
+      expect(cell(r, 1, 0)).toHaveTextContent("b");
+
+      // And when sorted by column 2
+      click(r.sortHeader_1);
+      // Then the `value: 2' row is first
+      expect(cell(r, 1, 0)).toHaveTextContent("b");
+
+      // And the rows were memoized so didn't re-render
+      expect(row(r, 1).getAttribute("data-render")).toEqual("1");
+      expect(row(r, 2).getAttribute("data-render")).toEqual("1");
+      expect(row(r, 3).getAttribute("data-render")).toEqual("1");
+    });
+
+    it("can sort priamry rows", async () => {
```

Comment:
Similar question, what's the difference between this test and the one on line 382?

Reply:
Old habit over providing the simplest implementation of a feature,  will remove 👍